tests/dslash_ctest.cpp

```diff
@@ -256,87 +169,53 @@ void init(int precision, QudaReconstructType link_recon) {
   }
 
+  csParam.setPrecision(inv_param.cpu_prec);
+  csParam.pad = 0;
 
- csParam.setPrecision(inv_param.cpu_prec);
- csParam.pad = 0;
-
- if(dslash_type == QUDA_DOMAIN_WALL_4D_DSLASH || dslash_type == QUDA_MOBIUS_DWF_DSLASH) {
-   csParam.siteSubset = QUDA_PARITY_SITE_SUBSET;
-   csParam.x[0] /= 2;
- } else {
-   if (test_type < 2 || test_type == 3) {
-     csParam.siteSubset = QUDA_PARITY_SITE_SUBSET;
-     csParam.x[0] /= 2;
-   } else {
-     csParam.siteSubset = QUDA_FULL_SITE_SUBSET;
-   }
- }
-
- csParam.siteOrder = QUDA_EVEN_ODD_SITE_ORDER;
- csParam.fieldOrder = QUDA_SPACE_SPIN_COLOR_FIELD_ORDER;
- csParam.gammaBasis = inv_param.gamma_basis;
- csParam.create = QUDA_ZERO_FIELD_CREATE;
-
- spinor = new cpuColorSpinorField(csParam);
- spinorOut = new cpuColorSpinorField(csParam);
- spinorRef = new cpuColorSpinorField(csParam);
- spinorTmp = new cpuColorSpinorField(csParam);
-
- csParam.x[0] = gauge_param.X[0];
-
- // printfQuda("Randomizing fields... ");
-
-
- //FIXME
- // if (strcmp(latfile,"")) {  // load in the command line supplied gauge field
- //   read_gauge_field(latfile, hostGauge, gauge_param.cpu_prec, gauge_param.X, argc, argv);
- //   construct_gauge_field(hostGauge, 2, gauge_param.cpu_prec, &gauge_param);
- // } else { // else generate a random SU(3) field
- construct_gauge_field(hostGauge, 1, gauge_param.cpu_prec, &gauge_param);
- // }
-
- spinor->Source(QUDA_RANDOM_SOURCE, 0);
+  if (dslash_type == QUDA_DOMAIN_WALL_4D_DSLASH || dslash_type == QUDA_MOBIUS_DWF_DSLASH) {
+    csParam.siteSubset = QUDA_PARITY_SITE_SUBSET;
+    csParam.x[0] /= 2;
+  } else {
+    if (test_type < 2 || test_type == 3) {
+      csParam.siteSubset = QUDA_PARITY_SITE_SUBSET;
+      csParam.x[0] /= 2;
+    } else {
+      csParam.siteSubset = QUDA_FULL_SITE_SUBSET;
+    }
+  }
 
- if (dslash_type == QUDA_CLOVER_WILSON_DSLASH || dslash_type == QUDA_TWISTED_CLOVER_DSLASH
-     || dslash_type == QUDA_CLOVER_HASENBUSCH_TWIST_DSLASH) {
-   double norm = 0.1; // clover components are random numbers in the range (-norm, norm)
-   double diag = 1.0; // constant added to the diagonal
-   construct_clover_field(hostClover, norm, diag, inv_param.clover_cpu_prec);
-   memcpy(hostCloverInv, hostClover, (size_t)V * cloverSiteSize * inv_param.clover_cpu_prec);
- }
+  csParam.siteOrder = QUDA_EVEN_ODD_SITE_ORDER;
+  csParam.fieldOrder = QUDA_SPACE_SPIN_COLOR_FIELD_ORDER;
+  csParam.gammaBasis = inv_param.gamma_basis;
+  csParam.create = QUDA_ZERO_FIELD_CREATE;
 
-  // printfQuda("done.\n"); fflush(stdout);
+  spinor = new cpuColorSpinorField(csParam);
+  spinorOut = new cpuColorSpinorField(csParam);
+  spinorRef = new cpuColorSpinorField(csParam);
+  spinorTmp = new cpuColorSpinorField(csParam);
 
-  // set verbosity prior to loadGaugeQuda
-  setVerbosity(verbosity);
-  inv_param.verbosity = verbosity;
+  csParam.x[0] = gauge_param.X[0];
 
-  // printfQuda("Sending gauge field to GPU\n");
+  constructHostGaugeField(hostGauge, gauge_param, argc_copy, argv_copy);
```

Comment:
might have missed it but where is copy initialized?

Reply:
found it, but what is the point of duplicating the pointer?

Reply:
Minor hangover from dslash_ctest.cpp. In that file, one needs to copy the args to global pointers so that they are visible to functions uses by Google Test. I took a quick look and, short of hacking the google test suite, this is the best option (in dslash_cstest.cpp) In this file, they are not needed. 8e1caf1ad